bubblewrap.c

```diff
@@ -2964,6 +3001,10 @@ main (int    argc,
       return monitor_child (event_fd, pid, setup_finished_pipe[0]);
     }
 
+  /* Unblock other signals here to receive signals from the parent. */
+  if (opt_signal_propogate)
+    gate_signals (SIG_UNBLOCK, &sigmask);
```

Comment:
If I'm understanding your intended design correctly, then this would probably be both clearer and more correct with:

```suggestion
  /* Unblock other signals here to receive signals from the parent. */
  if (opt_signal_propogate)
    {
      if (sigprocmask (SIG_SETMASK, sigmask_before_forwarding, NULL) != 0)
        die_with_error ("sigprocmask");
    }
```

This has two effects:

* inline just the single operation you want to do, which is to restore the signal mask that had been in effect before you blocked the signals that you want to forward, instead of having the rest of `gate_signals()` having to check for `SIG_UNBLOCK` and disable itself;
* use `SIG_SETMASK` instead of `SIG_UNBLOCK`, so that if a signal was already blocked before you called `gate_signals()`, it would remain blocked after this point, for least-astonishment